lib/routes/onAssessment.js

```diff
@@ -82,19 +82,20 @@ module.exports = [
                     mathMarksIn10th: Student.field("mathMarksIn10th"),
                     percentageIn12th: Student.field("percentageIn12th"),
                     mathMarksIn12th: Student.field("mathMarksIn12th"),
-                    typeOfTest: Joi.string().allow(CONSTANTS.typeOfTest).default("onlineTest")
                 }
             },
             handler: async (request, h) => {
                 const { assessmentService, studentService } = request.services();
 
                 let enrolmentKey = await internals.validateEnrolmentKey(assessmentService, request.params.enrolmentKey);
-
-                let payload = studentService.swapEnumKeysWithValues(request.payload);
-                await assessmentService.patchStudentDetails(enrolmentKey, payload);
-
-
-                return {'sucess': true}
+                // once student completed his test then no one given the test using same enrolment key.
+                if (enrolmentKey.student.stage != 'completedTest') {
+                    let payload = studentService.swapEnumKeysWithValues(request.payload);
+                    await assessmentService.patchStudentDetails(enrolmentKey, payload);
+                    return {'sucess': true}
+                }else{
+                    return {'Error': "This enrolment key is already used by another student. Please call to Navgurukuls  helpline number and take new test." }
```

Comment:
Please read about Boom and see how errors are thrown in our code base. Best if we maintain that standard.